enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,901 @@
+---
+title: declarative-index-config
+authors:
+  - "@anik120"
+reviewers:
+  - @njhale
+  - @benluddy
+  - @dmesser
+  - @exdx
+  - @Jamstah
+  - @bparees
+  - @ecordell
+approvers:
+  - @dmesser
+  - @bparees
+  - @ecordell
+  - @krizza
+creation-date: 2021-02-05
+last-updated: 2021-02-05
+status: implementable  
+---
+
+
+# Package representation and management in an index 
+
+### Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary 
+
+This enhancement proposes the representation of packages in an index in a standard declarative way(using json/yaml), so that index authors and operator authors can communicate with each other about the content of an index using json/yaml files, without needing to communicate with each other using containers(index images, operator bundle images etc). Using the declarative representation of the packages in the index, it also proposes allowing index authors and individual package owners to alter the properties of the packages(package name, channel information etc) as well as the structure of the packages (upgrade graphs of channels etc) of the index without having to rebuild bundles+republish the index. Finally, it discusses using the representations as source of truth for serving data from the index over a gRPC api, instead of the sqllite database that is used currently, along with migration strategy for existing indexes from the sqllite databases to the package representations.   
```

Comment:
We mention that the intention of this enhancement is to shift from building bundle and index images to using JSON and YAML files as the packaging format for Operators. Is there a TL;DR we can throw into this section that describes how we may use those JSON/YAML representations without having to read the rest of the enhancement?